internal/tofu/node_provider_eval.go

```diff
@@ -24,6 +25,19 @@ var _ GraphNodeExecutable = (*NodeEvalableProvider)(nil)
 
 // GraphNodeExecutable
 func (n *NodeEvalableProvider) Execute(ctx context.Context, evalCtx EvalContext, op walkOperation) (diags tfdiags.Diagnostics) {
-	_, err := evalCtx.InitProvider(ctx, n.Addr, addrs.NoKey)
-	return diags.Append(err)
+	provider, err := evalCtx.InitProvider(ctx, n.Addr, addrs.NoKey)
+
+	diags = diags.Append(err)
+	return diags.Append(n.InitEvalableProvider(ctx, evalCtx, provider))
```

Comment:
This is a tiny thing. But bothers me enough to mention.
I think calling `InitEvalableProvider` is better done on a separate line, and then appending the diagnostics.
Now, you might skim over it and not pay attention to it being passed as an argument to the `Append` function.
```suggestion
	provider, err := evalCtx.InitProvider(ctx, n.Addr, addrs.NoKey)
	diags = diags.Append(err)
	initDiags := n.InitEvalableProvider(ctx, evalCtx, provider)
	return diags.Append(initDiags)
```